src/lib.rs

```diff
@@ -102,87 +88,120 @@ pub fn demangle(mut s: &str) -> Demangle {
         });
 
         if all_hex {
-            s = &s[..i];
+            thinlto_stripped = &s[..i];
```

Comment:
Note a subtle change here: previously, this would strip the thinlto data from s before assigning original to it, which could lead to the removal of symbol data before passing it to original. This seemed to be erroneous.